menpofit/dpm/base.py

```diff
@@ -0,0 +1,1062 @@
+        elif 3 <= gmixid <= 9:
+            anno2tree = np.array(([34, 33, 32, 35, 36,
+                                   31, 30, 29, 28,
+                                   40, 41, 42, 39, 38, 37,
+                                   18, 19, 20, 21, 22,
+                                   43, 48, 47, 44, 45, 46,
+                                   27, 26, 25, 24, 23,
+                                   52, 51, 50, 49, 61, 62, 63, 53, 54, 55, 65, 64,
+                                   56, 66, 57, 67, 59, 68, 60, 58,
+                                   9,8 , 7, 6, 5, 4, 3, 2, 1, 10, 11, 12, 13, 14, 15, 16, 17]))
+        elif 10 <= gmixid <= 12:
+            anno2tree = np.array(([6, 5, 4, 3, 2, 1,
+                                   14, 15, 11, 12, 13,
+                                   10, 9, 8, 7,
+                                   16, 17, 18, 19, 20, 21, 22,
+                                   25, 24, 23, 26, 27,
+                                   28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39]))
+        else:
+            raise ValueError('No such model for anno2tree\' list exists.')
+        return anno2tree - 1  # -1 to change matlab indexes
+
+    @staticmethod
+    def _get_pie_image_info(pos_data, pos_data_dir, anno_dir, neg_data_dir):
+        pos = []
+        train_list = [50, 50, 50, 50, 50, 50, 300, 50, 50, 50, 50, 50, 50]
+        print('Collecting info for the positive images.')
+        for gmixid, poses in enumerate(pos_data):
+            count = 0
+            for img in poses['images']:
+                if count >= train_list[gmixid]:
+                    break
+                img_name = img[0][0]
+                [sub_id, ses_id, rec_id, cam_id, _] = img_name.split('_')
+                cam1_id = cam_id[0:2]
+                cam2_id = cam_id[2]
+                file_name = '{0}/session{1}/png/{2}/{3}/{4}_{5}/{6}.png'.format(pos_data_dir, ses_id, sub_id, rec_id, cam1_id, cam2_id, img_name)
+                anno_file_name = '{0}/{1}_lm.mat'.format(anno_dir, img_name)
+                if not os.path.isfile(file_name) or not os.path.isfile(anno_file_name):
+                    continue
+                count += 1
+                aux = dict()
+                aux['pts'] = scipy.io.loadmat(anno_file_name)['pts'][DPMLearner._get_anno2tree(gmixid), :]
+                aux['im'] = file_name
+                aux['gmixid'] = gmixid
+                pos.append(aux)
+        print('Collecting info for the negative images.')
+        l1 = sorted(os.listdir(neg_data_dir))
+        neg = []
+        for elem in l1:
+            if elem[elem.rfind('.') + 1:] in ['jpg', 'png', 'jpeg']:
+                aux = dict()
+                aux['im'] = os.path.join(neg_data_dir, elem)
+                neg.append(dict(aux))
+        return pos, neg
+
+    def _model_train(self):
+        multipie_mat = '/homes/ks3811/Phd/1 year/wildFace/face-release1.0-full/multipie.mat'
```

Comment:
Obviously we need to be able to set these paths as I don't have access your your directory and thus I can't even test this at the moment! I suppose maybe it would be good to put some of this data in your home dir on Atlas so we can test?